design/20210324.certificate-request-approve-deny.md

```diff
@@ -0,0 +1,191 @@
+---
+title: CertificateRequest Approve and Denied Conditions
+authors:
+  - "@joshvanl"
+editor: "@joshvanl"
+creation-date: 2021-02-03
+last-updated: 2021-02-03
+status: implementable
+---
+
+# Approved and Denied Conditions
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [API Behaviour](#api-behaviour)
+- [RBAC](#rbac)
+  * [Scope](#scope)
+  * [API Group](#api-group)
+  * [Resource](#resource)
+  * [Verbs](#verbs)
+  * [Resource Names](#resource-names)
+- [cert-manager Approver](#cert-manager-approver)
+- [Certificates Controller Changes](#certificates-controller-changes)
+  * [Issuing Controller](#issuing-controller)
+<!-- /toc -->
+
+## Summary
+
+The CertificateRequest resource has [Approved and Denied
+conditions](https://github.com/jetstack/cert-manager/blob/7204284063702358164713379b07b73642ea9bec/pkg/apis/certmanager/v1/types_certificaterequest.go#L194).
+These conditions are based upon the [certificates.k8s.io
+CertificateSigningRequest](https://github.com/kubernetes/api/blob/3a2d6b5bb7f84daa2125131865eb81b151448d5b/certificates/v1/types.go#L198)
+conditions of the same name. The purpose of these conditions is so that users,
+or "approvers", are able to mark a request as either Approved or Denied. This
+condition gates an Issuer from signing the request. An Issuer waits for a
+CertificateRequest to have an Approved condition before signing. A
+CertificateRequest with a Denied condition will never be signed.
+
+
+## API Behaviour
```

Comment:
Can we call out that the conditions are set via the `status` endpoint? It'd also be good to call out that this is a divergence from the upstream type, because we cannot create custom subresources.

Also worth noting that if a user *does not* have permission to approve, and attempts to approve the request (as well as updating other status fields), what will happen? Will the request be rejected by validation, or will the condition simply not be persisted? What does the upstream type do? I _think_ it just discards the changes (as this can be done rather simply due to the fact it is only correctly updated via the `/approval` subresource anyway)